hazelcast/src/test/java/com/hazelcast/osgi/HazelcastOSGiServiceTest.java

```diff
@@ -0,0 +1,389 @@
+            System.setProperty(HazelcastOSGiService.HAZELCAST_OSGI_GROUPING_DISABLED, "true");
+
+            testBundle = new TestBundle();
+            TestBundleContext testBundleContext = testBundle.getBundleContext();
+
+            testBundle.start();
+
+            HazelcastInternalOSGiService service = getService(testBundleContext);
+
+            Config config = new Config();
+            config.getGroupConfig().setName(GROUP_NAME);
+
+            HazelcastOSGiInstance osgiInstance = service.newHazelcastInstance(config);
+            assertNotNull(osgiInstance);
+            assertEquals(GROUP_NAME, osgiInstance.getConfig().getGroupConfig().getName());
+
+            HazelcastInstance instance = osgiInstance.getDelegatedInstance();
+            assertNotNull(instance);
+            assertEquals(GROUP_NAME, instance.getConfig().getGroupConfig().getName());
+        } finally {
+            if (propValue != null) {
+                System.setProperty(HazelcastOSGiService.HAZELCAST_OSGI_GROUPING_DISABLED, propValue);
+            }
+            if (testBundle != null) {
+                testBundle.stop();
+            }
+        }
+    }
+
+    @Test
+    public void instanceRetrievedSuccessfullyWithItsName() {
+        final String INSTANCE_NAME = "test-osgi-instance";
+
+        HazelcastInternalOSGiService service = getService();
+
+        Config config = new Config(INSTANCE_NAME);
+
+        service.newHazelcastInstance(config);
+
+        HazelcastOSGiInstance osgiInstance = service.getHazelcastInstanceByName(INSTANCE_NAME);
+        assertNotNull(osgiInstance);
+        assertEquals(config, osgiInstance.getConfig());
+
+        HazelcastInstance instance = osgiInstance.getDelegatedInstance();
+        assertNotNull(instance);
+        assertEquals(config, instance.getConfig());
+    }
+
+    @Test
+    public void allInstancesRetrievedSuccessfully() {
+        Set<HazelcastOSGiInstance> osgiInstances = new HashSet<HazelcastOSGiInstance>();
+
+        HazelcastInternalOSGiService service = getService();
+
+        osgiInstances.add(service.newHazelcastInstance());
+        osgiInstances.add(service.newHazelcastInstance(new Config("test-osgi-instance")));
+
+        Set<HazelcastOSGiInstance> allOSGiInstances = service.getAllHazelcastInstances();
+        assertEquals(osgiInstances.size(), allOSGiInstances.size());
+        assertTrue(allOSGiInstances.containsAll(osgiInstances));
+    }
+
+    @Test
+    public void instanceShutdownSuccessfully() {
+        final String INSTANCE_NAME = "test-osgi-instance";
+
+        HazelcastInternalOSGiService service = getService();
+
+        Config config = new Config(INSTANCE_NAME);
+
+        HazelcastOSGiInstance osgiInstance = service.newHazelcastInstance(config);
+        assertNotNull(osgiInstance);
+        assertTrue(osgiInstance.getLifecycleService().isRunning());
+
+        HazelcastInstance instance = osgiInstance.getDelegatedInstance();
+        assertNotNull(instance);
```

Comment:
This assert should not be needed. If you don't believe that you got back a valid instance, you should cover this in another test. The same goes for the asserts above.

The only asserts which should be in your tests are the ones below service.shutdown...